silver/tests/spec/test_payment_method.py

```diff
@@ -1,49 +1,330 @@
+
+        return method
+
+    def test_get_listing(self):
+        PaymentMethodFactory.create(customer=CustomerFactory.create())
+        method = self.create_payment_method(customer=self.customer)
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        self.assert_get_data(url, [method])
+
+    def test_get_detail(self):
+        PaymentMethodFactory.create(customer=CustomerFactory.create())
+        method = self.create_payment_method(customer=self.customer)
+
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': self.customer.pk,
+            'payment_method_id': method.pk
+        })
+
+        self.assert_get_data(url, method)
+
+    def test_post_listing(self):
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        response = self.client.post(url, data={
+            'payment_processor': processor_url,
+            'state': PaymentMethod.States.Uninitialized
+        }, format='json')
+
+        method = PaymentMethod.objects.get(customer=self.customer)
+        self.assert_get_data(response.data['url'], method)
+
+    def test_post_listing_additional_data_wrong_state(self):
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        response = self.client.post(url, data={
+            'payment_processor': processor_url,
+            'state': PaymentMethod.States.Uninitialized,
+            'additional_data': '{"random": "value"}'
+        }, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_post_listing_invalid_initial_state(self):
+        invalid_initial_states = list(
+            set([s[0] for s in PaymentMethod.States.Choices]) -
+            {PaymentMethod.States.Uninitialized,
+             PaymentMethod.States.Unverified,
+             PaymentMethod.States.Enabled})
+
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        for state in invalid_initial_states:
+            response = self.client.post(url, data={
+                'payment_processor': processor_url,
+                'state': state
+            }, format='json')
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_detail_state_transitions(self):
+        states = [x[0] for x in PaymentMethod.States.Choices]
+        permutations = [(old, new) for old in states for new in states]
```

Comment:
No need for another import just for that one line alone.
And yes there is even a better solution than permutations i need the `itertools.product`